cmd/scale.go

```diff
@@ -255,16 +288,35 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 		currentNodeCount = len(indexes)
 
 		if currentNodeCount == sc.newDesiredAgentCount {
-			log.Info("Cluster is currently at the desired agent count.")
+			sc.printScaleTargetEqualsExisting(currentNodeCount)
 			return nil
 		}
 		highestUsedIndex = indexes[len(indexes)-1]
 
-		// Scale down Scenario
+		// VMAS Scale in Scenario
 		if currentNodeCount > sc.newDesiredAgentCount {
-			if sc.masterFQDN == "" {
+			if sc.apiserverURL == "" {
 				cmd.Usage()
-				return errors.New("master-FQDN is required to scale down a kubernetes cluster's agent pool")
+				return errors.New("--apiserver (or the deprecated --master-FQDN) is required to scale down a kubernetes cluster's agent pool")
+			}
+
+			if sc.nodes != nil {
+				if len(sc.nodes) > 1 {
+					sc.logger.Infof("There are %d nodes in pool %s before scaling down to %d:\n", len(sc.nodes), sc.agentPoolToScale, sc.newDesiredAgentCount)
+				} else {
+					sc.logger.Infof("There is %d node in pool %s before scaling down to %d:\n", len(sc.nodes), sc.agentPoolToScale, sc.newDesiredAgentCount)
+				}
+				operations.PrintNodes(sc.nodes)
+				numNodesFromK8sAPI := len(sc.nodes)
+				if currentNodeCount != numNodesFromK8sAPI {
+					sc.logger.Warnf("The Azure API reports %d vms in pool %s", currentNodeCount, sc.agentPoolToScale)
```

Comment:
what is the "Azure API" in this case? As a user it might not be clear to me what this warning means. Maybe something like this would make more sense: `There are %d VMs in pool %s but K8s is tracking %s nodes", currentNodeCount, sc.agentPoolToScale, numNodesFromK8sAPI `

Reply:
Mades some changes